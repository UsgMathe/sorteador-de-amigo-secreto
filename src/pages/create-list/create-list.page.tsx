import { Card } from '../../components/card';
import { Footer } from '../../components/footer/footer';
import { Form } from '../../components/form/form';

export function CreateListPage() {
  return (
    <Card>
      <div className="flex flex-col max-w-6xl grow w-full mx-auto space-y-10">
        <Form />

        <div className="mt-auto flex items-center justify-center gap-12 w-full flex-wrap items-center">
          <Footer />
          <img
            src="/sacolas.png"
            alt="sacolas"
            className="max-w-[10rem] md:max-w-[12rem]"
          />
        </div>
      </div>
    </Card>
  );
}
