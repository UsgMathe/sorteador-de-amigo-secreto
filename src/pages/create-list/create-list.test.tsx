import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import { CreateListPage } from './create-list.page';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

describe('a pagina de criação de lista de participantes', () => {
  test('deve ser renderizada corretamente', () => {
    const { container } = render(
      <RecoilRoot>
        <CreateListPage />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
