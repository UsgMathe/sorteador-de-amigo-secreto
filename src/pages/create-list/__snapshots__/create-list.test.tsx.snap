// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de criação de lista de participantes deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="flex flex-col min-h-screen"
  >
    <header>
      <div
        class="p-20 sm:mb-4 mx-auto sm:mx-0"
      >
        <img
          alt="logo Sorteador de Amigo Secreto"
          class="w-fit hidden sm:block mb-20"
          src="/logo.png"
        />
        <img
          alt="logo Sorteador de Amigo Secreto"
          class="w-fit mx-auto sm:hidden mb-20"
          src="/logo-pequeno.png"
        />
      </div>
    </header>
    <div
      class="grow flex flex-col h-full bg-background p-15 border-t-4 border-x-4 rounded-t-[5rem] relative"
    >
      <div
        class="absolute sm:-top-53 -top-43 z-10 w-full flex justify-center sm:justify-end px-10 left-0"
      >
        <img
          alt="Ilustração de personagem fictício"
          class="sm:w-96 w-78 flex-none"
          src="/person.png"
        />
      </div>
      <div
        class="flex flex-col max-w-6xl grow w-full mx-auto space-y-10"
      >
        <form
          class="flex flex-col items-center space-y-10 "
        >
          <div
            class="space-y-8"
          >
            <h1
              class="text-3xl text-center text-secondary font-semibold font-poppins"
            >
              Vamos começar!
            </h1>
            <div
              class="flex flex-col"
            >
              <label
                aria-invalid="false"
                class="rounded-4xl border-2 flex items-center overflow-hidden space-x-8 pl-8 max-w-4xl bg-white w-full input-shadow aria-invalid:border-red-500"
                for="persons-input"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-user-plus flex-none fill-muted-foreground text-muted-foreground size-5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                  />
                  <circle
                    cx="9"
                    cy="7"
                    r="4"
                  />
                  <line
                    x1="19"
                    x2="19"
                    y1="8"
                    y2="14"
                  />
                  <line
                    x1="22"
                    x2="16"
                    y1="11"
                    y2="11"
                  />
                </svg>
                <input
                  class="w-full py-4 focus-within:outline-0 text-sm overflow-ellipsis"
                  id="persons-input"
                  placeholder="Insira os nomes dos participantes"
                  type="text"
                  value=""
                />
                <button
                  class="hidden sm:block rounded-r-2xl border-l text-sm py-5 px-12 cursor-pointer disabled:bg-muted-background text-white disabled:text-button-text hover:bg-secondary hover:text-white transition-colors duration-300 disabled:pointer-events-none disabled:opacity-70 select-none bg-secondary hover:opacity-95"
                  data-testid="desktop-button"
                  disabled=""
                >
                  Adicionar
                </button>
              </label>
              
              <button
                class="button-shadow border-2 mt-4 mx-auto rounded-4xl border-l text-sm py-5 px-12 cursor-pointer transition-all duration-300 disabled:pointer-events-none disabled:opacity-70 select-none border-black flex items-center gap-2 hover:opacity-95 hover:scale-[98%] bg-secondary sm:hidden disabled:bg-muted-background disabled:text-button-text text-white"
                data-testid="mobile-button"
                disabled=""
              >
                Adicionar
              </button>
            </div>
          </div>
          <div
            class="mt-4"
          >
            <h2
              class="mb-4 border-b-2 w-fit mx-auto"
            >
              Participantes
            </h2>
            <p
              class="text-muted-foreground"
            >
              Sem participantes
            </p>
          </div>
        </form>
        <div
          class="mt-auto flex items-center justify-center gap-12 w-full flex-wrap items-center"
        >
          <footer>
            <button
              class="button-shadow border-2 mt-4 mx-auto rounded-4xl border-l text-sm py-5 px-12 cursor-pointer disabled:pointer-events-none disabled:opacity-70 select-none border-black flex items-center gap-2 bg-primary text-white hover:opacity-95 transition-all duration-300 hover:scale-[98%] disabled:bg-muted-background"
              disabled=""
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-circle-play"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <polygon
                  points="10 8 16 12 10 16 10 8"
                />
              </svg>
              Iniciar Sorteio!
            </button>
          </footer>
          <img
            alt="sacolas"
            class="max-w-[10rem] md:max-w-[12rem]"
            src="/sacolas.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
